### ISSUE TITLE
Support custom file suffixes/prefixes/names

### ISSUE THREAD

**Issue by deniszholob** [cyclosproject/ng-openapi-gen#200]
Does the tool support adding custom suffixes like attaching a `.model.ts` or `.enum.ts` to the end of every interface/enum generated?
If not, can this be added in?

**Comment by luisfpg** [comment]
Top-level enums are handled as models.
You can set custom prefix / suffix for models or services in the configuration file.

**Comment by luisfpg** [comment]
No.
Also, take into account that it's not just renaming a file.
The class name is renamed accordingly too, and it affects all imports and references from other models and services.

**Comment by deniszholob** [comment]
I wasnt really trying to rename the classes, shouldnt be affected by the filename right, i would have assumed both are driven by the api.yml itself?
I was more thinking as its generating the filenames/imports it would just rename/update the path as defined in the regex config thats all.
Maybe im not realizing the complexity as im not sure about the internal workings.

**Comment by luisfpg** [comment]
"The Angular way" states that class `HelloIamAClass` should be placed in the file named `hello-i-am-a-class.ts`.
So, renaming just the file name would break this.
Also the file name is explicitly referenced when others import the class (other models or services).
Also, trust me, configuring the generator by "just adding a random regex here or there" is not simple at all.

**Comment by deniszholob** [comment]
Makes sense, ive tried to use the prefix config with `"modelSuffix":".model"` and it took my `.` and replaced that with a `-` :(
I get you are trying to stick to the angular convention but it also has the `.services`, `.pipes` etc... suffixes which is what i am trying to do with the models.
Not a huge deal in the long run, but a nice to have.

**Comment by luisfpg** [comment]
Actually, the model suffix is for the class name.
The file name will be derived from it.
So, if you have `"modelSuffix": "Model"`, your model `User` will be renamed to `UserModel`, and the corresponding file will be `user-model.ts`.

**Comment by deniszholob** [comment]
Ohhh, so we misunderstood one another, so then my original question is then valid still. Attaching suffixes to file only, not the model name itself.

**Comment by gnitter** [comment]
It's an old question, but i think it still would be a great feature, if it's possible to attach custom suffixes to generated files. E.g. `file.model.ts`.
